code/modules/keybindings/bindings_atom.dm

```diff
@@ -0,0 +1,37 @@
+// You might be wondering why this isn't client level. If focus is null, we don't want you to move.
+// Only way to do that is to tie the behavior into the focus's keyLoop().
+
+/atom/movable/keyLoop(client/user)
+	if(!user.keys_held["Ctrl"] && isturf(loc))
+		var/movement_dir = null
+		var/list/movement = SSinput.movement_keys
+		for(var/_key in user.keys_held)
+			movement_dir |= movement[_key]
+		if(user.next_move_dir_add)
+			movement_dir |= user.next_move_dir_add
+		if(user.next_move_dir_sub)
+			movement_dir &= ~user.next_move_dir_sub
```

Comment:
I don't get what this does. Why is using held keys not sufficient (note that these are tracking recently pressed held keys)?